cairo rm v31 integration

XuyangSong opened the pull request:
based on #1912

This PR can replace #1897 and #1813

XuyangSong committed:
modify shielded transaction and action structures

XuyangSong committed:
rename shielded_transaction.ex to transaction.ex

XuyangSong committed:
cache more cairo examples to boost up CI tests

XuyangSong committed:
rename delta to delta_proof in cairo transaction

XuyangSong committed:
rename compliance_proofs to compliance_units in cairo action

XuyangSong committed:
use Map for compliance_units in the cairo action

agureev committed:
Make Node Scry Nounify Results

agureev committed:
Add Client Example of Getting a Nounified Result from Scry

agureev committed:
Make Sure Client-Side Reads are Nouns

XuyangSong committed:
use Map for actions in Cairo Transaction

XuyangSong committed:
add logic_hash to logic_proofs struct in Cairo Action

XuyangSong committed:
add app_data to action in Cairo RM

XuyangSong committed:
store cairo app_data to blob

XuyangSong committed:
refactor cairo ProofRecord

XuyangSong committed:
rename public_inputs to instance in cairo rm

XuyangSong committed:
rename compliance_inputs to compliance_witness in cairo rm

XuyangSong committed:
rename circuit to proving key in cairo rm

XuyangSong committed:
add prove_delta method to cairo transactioin

XuyangSong committed:
remove Anoma.RM.Transaction interfaces

XuyangSong committed:
refine cairo tx and action

XuyangSong committed:
rename logic to logic_ref in cairo rm

XuyangSong committed:
rename label to label_ref in cairo rm

XuyangSong committed:
rename data to value_ref in cairo rm

XuyangSong committed:
rename eph to is_ephemeral in cairo rm

XuyangSong committed:
rename rseed to randseed in cairo rm

XuyangSong committed:
rename private_input to witness in cairo rm examples

XuyangSong committed:
update names in cairo circuits

XuyangSong committed:
add is_consumed flag to resource logic in cairo rm

XuyangSong committed:
fix cairo tx examples

agureev committed:
Add DPS Nounable Protocol

agureev committed:
Add Resource Nounable Protocol

agureev committed:
Add CPS Nounable Protocol

agureev committed:
Add RLPS Nounable Protocol

agureev committed:
Add CU Nounable Protocol

agureev committed:
Add Action Nounable Protocol

agureev committed:
Add Transaction Nounable Protocol

agureev committed:
Merge branch 'artem/bug-nounable-list' into artem/trm-extra-api

agureev committed:
Merge branch 'artem/trm-nounable' into artem/trm-extra-api

agureev committed:
Merge branch 'artem/bug-deltahash-fix' into artem/trm-extra-api

agureev committed:
Merge branch 'artem/bug-commitmentaccumulator-fix' into artem/trm-extra-api

agureev committed:
Merge branch 'artem/bug-resource-fix' into artem/trm-extra-api

agureev committed:
Merge branch 'artem/bug-cups-fix' into artem/trm-extra-api

agureev committed:
Merge branch 'artem/bug-cu-fix' into artem/trm-extra-api

agureev committed:
Merge branch 'artem/bug-action-fix' into artem/trm-extra-api

agureev committed:
Add Additional Resource API

Provide usual Resource API: propositions testing commitment or nullifier

agureev committed:
Root Computations for Compliance Units

Provide a function to gather compliance unit roots.

agureev committed:
API for Actions

Provides API to get roots and app data from the action.

agureev committed:
Add Transaction API

Add transaction API, including getting all roots, all commitments, all
nullifiers, or all app data from th underlying actions.

agureev committed:
Add Transaction Protocol API for TRM V2 Transaction

XuyangSong committed:
add the padding resource

XuyangSong committed:
remove unused code in cairo rm

agureev commented on apps/anoma_node/lib/node/transaction/backends.ex:
Blob storage is content-adressed, so we need to use the sha256 value as the key instead of the tag.

agureev reviewed (changes requested):
App data blob storage needs change.

XuyangSong commented on apps/anoma_node/lib/node/transaction/backends.ex:
fixed

agureev committed:
Fix VK Mug Merge Bug

Wrong parent computation.

agureev committed:
Merge remote-tracking branch 'origin/artem/nock-nounify-scry' into artem/trm-v2-integration

agureev committed:
Merge remote-tracking branch 'origin/artem/trm-extra-api' into artem/trm-v2-integration

agureev committed:
Merge branch 'artem/bug-merge-mugs' into artem/trm-v2-integration

agureev committed:
Integrate TRMV2 Into the Node

Remove all usage of previous transparent resource machine usage from
the codebase.

agureev committed:
make roots in shielded tx a set
